kolibri/core/assets/src/views/CoreMenu/CoreMenuOption.vue

```diff
@@ -178,6 +186,10 @@
       handleNav(route) {
         this.isActiveLink(route) ? this.toggleAndroidMenu() : null;
       },
+      generateNavRoute(route) {
+        console.log('url', this.link, 'route', route, 'subroutes', this.subRoutes);
+        return generateNavRoute(this.link, route, this.subRoutes);
```

Comment:
Not sure why we need to pass in all the subroutes here - we have the route object for this specific subroute, it doesn't need to know about the others. I think cleaning that up will simplify the logic in the generateNavRoute function too.

The other thing is - do we want to conditionalize this - if we are in `this.link` we can use vue router to generate the link, if not we need to use the route generation.

This is also the place where we could reinject params from the current route if we are on the SPA in question.